dpgen/data/gen.py

```diff
@@ -237,12 +237,24 @@
 def poscar_scale(poscar_in, poscar_out, scale):
     with open(poscar_in) as fin:
         lines = list(fin)
-    if "D" == lines[7][0] or "d" == lines[7][0]:
+
+    # Determine if "Selective dynamics" is present
+    if lines[7].strip().lower().startswith("s"):
```

Comment:
It's a good practice to use constants for magic numbers like 7 and 8 that appear multiple times in your code. This will make the code more maintainable and easier to understand. [medium]